src/components/Form.js

```diff
@@ -17,32 +16,46 @@ export default class Form extends Component {
     noValidate: false,
     liveValidate: false,
     safeRenderCompletion: false,
-  }
+  };
 
   constructor(props) {
     super(props);
     this.state = this.getStateFromProps(props);
   }
 
+  componentDidMount() {
+    const {formData} = this.state;
+    const {liveValidate, noValidate, schema} = this.props;
+    const {errors = [], errorSchema = {}} = this.state;
+    const mustValidate = !!formData && !noValidate && liveValidate;
+
+    Promise
```

Comment:
How about we introduce usage of async/await? That should really help reading the flows in this patch. Should be matter of enabling it in .babelrc.

Reply:
i know this is four years later, ok no i don't care that it's four years later... but

promises still have their place. not everything is async/await suitable.